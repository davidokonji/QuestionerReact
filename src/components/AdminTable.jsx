import React from 'react';
import { connect } from 'react-redux';
import MeetupList from './MeetupList';
import { deleteMeetup } from '../actions';

const AdminTable = ({ data, deleteOne }) => (
  <div className='col-11 mx-4 mx-md-0 mt-4 mt-md-0 col-md-9 shadow p-0 table-responsive-sm mb-5'>
    <table className='table table-hover p-0'>
      <thead className='row mx-0'>
        <div className='col-10 col-md-10 bg-light p-3'>
            Available Meetup(s)
        </div>
        <div className='col-2 col-md-2 bg-light p-3'>
            Operation
        </div>
      </thead>
      <tbody className='row p-2 col-12 ml-0 '>
        {
          data.map(meetup => (
            <MeetupList
              id={meetup.id}
              title={meetup.title}
              location={meetup.location}
              tags={meetup.tags}
              happeningOn={meetup.happeningOn}
              images={meetup.images}
              key={meetup.id}
              deleteOne={deleteOne}
            />
          ))
      }
      </tbody>
    </table>
  </div>
);
const mapDispatchToProps = {
  deleteOne: deleteMeetup
};

const AdminDash = connect(null, mapDispatchToProps)(AdminTable);

export {
  AdminDash,
  AdminTable
};
