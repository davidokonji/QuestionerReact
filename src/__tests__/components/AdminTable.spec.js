import React from 'react';
import { shallow } from 'enzyme';
import AdminTable from '../../components/AdminTable';

describe('Admin table component', () => {
  it('should render component correctly', () => {
    const toggle = jest.fn();
    const data = [];
    const wrapper = shallow(<AdminTable toggle={toggle} data={data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
