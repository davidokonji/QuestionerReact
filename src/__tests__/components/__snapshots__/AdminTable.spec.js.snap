// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin table component should render component correctly 1`] = `
<div
  className="col-11 mx-4 mx-md-0 mt-4 mt-md-0 col-md-9 shadow p-0 table-responsive-sm mb-5"
>
  <table
    className="table table-hover p-0"
  >
    <thead
      className="row mx-0"
    >
      <div
        className="col-10 col-md-10 bg-light p-3"
      >
        Available Meetup(s)
      </div>
      <div
        className="col-2 col-md-2 bg-light p-3"
      >
        Operation
      </div>
    </thead>
    <tbody
      className="row p-2 col-12 ml-0 "
    />
  </table>
</div>
`;
